vars/buildPlugin.groovy

```diff
@@ -173,5 +195,80 @@ exit \$PUBLISHED
 		}
 	}
 
+	tasks["ubuntu-ppa"] = {
+		throttle(["binary-addons/ubuntu-ppa-${version}"])
+		{
+			node("linux")
+			{
+				ws("workspace/binary-addons/kodi-ubuntu-ppa-${version}")
+				{
+					stage("clone ubuntu-ppa")
+					{
+						dir("${addon}")
+						{
+							if (env.BRANCH_NAME)
+							{
+								def scmVars = checkout(scm)
+								currentBuild.displayName = scmVars.GIT_BRANCH + '-' + scmVars.GIT_COMMIT.substring(0, 7)
+							}
+							else if ((env.BRANCH_NAME == null) && (repo))
+							{
+								git repo
+							}
+							else
+							{
+								error 'buildPlugin must be used as part of a Multibranch Pipeline *or* a `repo` argument must be provided'
+							}
+						}
+					}
+
+					stage("build ubuntu-ppa")
+					{
+						if (params.force_ppa_upload)
+						{
+							sh "rm -f kodi-*.changes kodi-*.build kodi-*.upload"
+						}
+
+						dir("${addon}")
+						{
+							echo "Ubuntu dists enabled: ${dists} - TAGREV: ${params.TAGREV} - PPA: ${params.PPA}"
```

Comment:
This is only a echo, but wouln't the resolved ppa (`${ppa}`) be better then auto.

Reply:
you see the actual ppa url later